### ISSUE TITLE
[explicit-module-boundary-types] ignore unused parameters

### ISSUE THREAD

**Issue by vegerot** [typescript-eslint/typescript-eslint#2311]
**Repro**

```typescript
export const test = (param1: string, _param2) => console.log(param1)
```

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-module-boundary-types": ['error', {
           ignore-unused-parameters: true
       }
  }
}
```

```TS
// your repro code case
export const test = (param1: string, _param2) => console.log(param1)
```

**Expected Result**
no error
**Actual Result**
```
Argument '_param2' should be typed.eslint@typescript-eslint/explicit-module-boundary-types
```
**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `latest` |
| `@typescript-eslint/parser`        | `latest` |
| `TypeScript`                       | `latest` |
| `ESLint`                           | `latest` |
| `node`                             | `X.Y.Z` |
| `npm`                              | `X.Y.Z` |

**Comment by bradzacher** [comment]
Thanks for the request! I'm going to reject this proposal for a few reasons:
- detecting unused variables is a pain to do.
- having unused code is just cruft that shouldn't be left around.
- having untyped parameters is easily avoided with a simple `unknown` annotation.

**Comment by vegerot** [comment]
what about ignoring parameters starting with `_`?
